Add io.github.kalaksi.Lightkeeper

kalaksi opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

hfiguiere commented on io.github.kalaksi.Lightkeeper.yml:
```diff
@@ -0,0 +1,58 @@
+      # Remove one patch that modifies configure.ac to avoid needing aclocal-1.14.
+      - rm debian-package/patches/0002-autoupdate-and-enable-automake-Wall.patch
+      - for patchfile in debian-package/patches/*.patch; do patch -p1 < "$patchfile"; done
+      - ./configure --without-perl-bindings --prefix=/app
```
this should be a `buildsytem: autotools` build. you can add a shell to do some pre-processing.

hfiguiere commented on io.github.kalaksi.Lightkeeper.yml:
```diff
@@ -0,0 +1,58 @@
+build-options:
+    append-path: /usr/lib/sdk/rust-stable/bin
+    build-args:
+      - "--share=network"
```
this is not allowed and will fail. build MUST be done offline.

hfiguiere commented:
bot, build io.github.kalaksi.Lightkeeper

hfiguiere commented:
```
builddir/files/share/appdata/io.github.kalaksi.Lightkeeper.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• style-invalid         : Not enough <screenshot> tags, minimum is 1
• tag-missing           : <release> required
• tag-missing           : <url> is not present
Validation of files failed
```

kalaksi commented:
I updated the appdata file and used appstream-util to verify it, so now it should be OK.

hfiguiere commented:
bot, build io.github.kalaksi.Lightkeeper

barthalion commented:
```
~❱ flatpak run io.github.kalaksi.Lightkeeper
[2023-07-10T06:41:29Z ERROR lightkeeper] Error while reading configuration file: No such file or directory (os error 2)
```

barthalion commented:
The icon also doesn't show up in my GNOME and uses the generic one.

barthalion commented:
Just to clarify, it's not a problem it requires some configuration before the first launch, but users should be told where it is, and probably some explanation how to configure anything if some template with comments isn't copied there.

kalaksi commented:
Configuration and file locations are currently explained in the repo's README, but I'm likely soon going to implement a dialog for first launch to help with all that.

hfiguiere commented:
> Configuration and file locations are currently explained in the repo's README,

No user ever see this.

kalaksi commented:
It now copies the default configuration on start, if configuration is missing, and displays a notification that advices user to take a look at the project's page for configuration help.

I also tried to resolve the icon issue and noticed it was being copied to a wrong location. There was a warning about icon not being exported. I fixed it but can't seem to get it to display the icon locally even though I also added icon for size 64x64 too. It doesn't display any warning for me anymore, so I'm not sure if it's some weird issue with my system or what.

hfiguiere commented:
bot, build io.github.kalaksi.Lightkeeper

barthalion commented:
/merge

kalaksi commented:
How do we proceed? I haven't yet received the write access invitation for https://github.com/flathub/io.github.kalaksi.Lightkeeper

barthalion commented:
Yes, there's some bug that prevents the automation from inviting developers. This should be done now.
